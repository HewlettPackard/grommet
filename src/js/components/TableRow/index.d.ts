import * as React from "react";

export interface TableRowProps {
  
}

declare const TableRow: React.StatelessComponent<TableRowProps>;

export { TableRow };
