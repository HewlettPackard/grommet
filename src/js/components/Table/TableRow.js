import React from 'react';

import { StyledTableRow } from './StyledTable';

const TableRow = props => <StyledTableRow {...props} />;

export default TableRow;
