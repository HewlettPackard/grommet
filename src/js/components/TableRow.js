// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import classnames from 'classnames';

const CLASS_ROOT = 'table-row';

const TableRow = props => {
  let classes = classnames(
    CLASS_ROOT,
    props.className,
    {
      [`${CLASS_ROOT}--selected`]: props.selected,
      [`${CLASS_ROOT}--selectable`]: props.onClick
    }
  );

  return (
    <tr className={classes} onClick={props.onClick}>
      {props.children}
    </tr>
  );
};

TableRow.propTypes = {
  onClick: PropTypes.func,
  selected: PropTypes.bool
};

TableRow.displayName = 'TableRow';

export default TableRow;
