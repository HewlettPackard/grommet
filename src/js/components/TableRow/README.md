## TableRow
A row of cells in a table.

## Usage

```javascript
import { TableRow } from 'grommet';
<TableRow />
```

## Properties

**hoverIndicator**

Whether to apply the hover indicator when the user is mousing over row.

```
boolean
```
  
## Intrinsic element

```
tr
```