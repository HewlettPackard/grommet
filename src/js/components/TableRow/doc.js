import { describe, PropTypes } from 'react-desc';

export const doc = TableRow => {
  const DocumentedTableRow = describe(TableRow)
    .description('A row of cells in a table.')
    .usage(
      `import { TableRow } from 'grommet';
<TableRow />`,
    )
    .intrinsicElement('tr');

    DocumentedTableRow.propTypes = {
      hoverIndicator: PropTypes.bool
        .description(
          'Whether to apply the hover indicator when the user is mousing over row.'
        )
        .defaultValue(false),
    }

  return DocumentedTableRow;
};
