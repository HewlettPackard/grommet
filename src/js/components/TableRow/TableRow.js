import React from 'react';

import { StyledTableRow } from '../Table/StyledTable';
import { doc } from './doc';

const TableRow = props => <StyledTableRow {...props} />;

const TableRowWrapper = process.env.NODE_ENV !== 'production' ? doc(TableRow) : TableRow;

export { TableRowWrapper as TableRow };

