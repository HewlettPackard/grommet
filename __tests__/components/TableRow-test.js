// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import TableRow from '../../src/js/components/TableRow';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('TableRow', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <TableRow custom={true} disabled={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
