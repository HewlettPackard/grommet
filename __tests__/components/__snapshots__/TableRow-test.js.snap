exports[`TableRow has correct default options 1`] = `
<tr
  className="grommetux-table-row grommetux-table-row--disabled"
  disabled={true}
  onClick={null}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]} />
`;
